docs/rfcs/XeGPU.md

```diff
@@ -701,6 +697,334 @@ An example on how to perform transpose using load with chunk_size in SIMT flavor
+  * wi_layout: Defines the 2D arrangement of WIs within the subgroup.
+  * wi_data: Specifies the shape of the tensor fragment that each WI loads or stores as a single packed data unit (16/32-bit).
+  * order: The dimension order used to linearize n-d subgroup ids to 1-d. The first dimension in the order list is the fastest-changing dimension.
+
+Example of linerized subgourp id regarding order[1, 0] vs. order [0, 1]. 
+| sg_layout[4, 4] | order[1, 0] | order[0, 1]
+| :----   | :----   |  :----   |
+| [0, 0], [0, 1], [0, 2], [0, 3] | 0, 1, 2, 3 | 0, 4, 8, 12 |
+| [1, 0], [1, 1], [1, 2], [1, 3] | 4, 5 , 6, 7| 1, 5, 9, 13 |
+| [2, 0], [2, 1], [2, 2], [2, 3] | 8, 9, 10 , 11 | 2, 6, 10, 14 |
+| [3, 0], [3, 1], [3, 2], [3, 3] | 12, 13, 14, 15 | 3, 7, 11, 15 |
+
+For a subgroup in 3-d sg_layout [dim_0, dim_1, dim_2], order[2, 1, 0] maps a subgroup with 3-d index [x, y, z] to a linear subgroup index [z + dim_2 * y + dim_2 * dim_1 * x ], order[1, 2, 0] maps to [y + dim_2 * z + dim_2 * dim_1 * x].
+
+When a sg_map attribute is attached to a tensor descriptor, load/store/dpas will operate at the workgroup level. The sg_map attribute must be specified when creating the tensor descriptor.
+
+**Constraints**
+
+Given these definitions:
+```mlir
+sg_data_size = sg_data[0] × sg_data[1]
+workgroup_size = sg_layout[0] × sg_layout[1]
+tensor_size = tensor_desc[0] × tensor_desc[1]
+```
+
+The following conditions must hold:
+
+* workgroup_size must represent the number of subgroups in a workgroup for a kernel.
+* tensor_desc[0] must be either evenly divisible by sg_layout[0] × sg_data[0], or vice versa.
+* tensor_desc[1] must be either evenly divisible by sg_layout[1] × sg_data[1], or vice versa.
+
+**distribution rule**
+
+The tensor_desc is distributed to sg_data x sg_layout along each dimension in a round-robin fashion. If sg_data[i] x sg_layout[i] < tensor_desc[i], there is data left after all subgroups are assigned for the first round, the rest data will wrap around and be assigned to the first subgroup until the data is completely assigned. If sg_data[i] x sg_layout[i] > tensor_desc[i], the data may be used up before all subgroups are assigned. In this case, we broadcast the tensor data to multiple subgroups by repeating the data assignment to the rest subgroups along that dimension until the all subgroups get data.
+
+**Resulting subgroup Data size**
+
+The distributed tensor for each subgroup has the same dimension as the work group level tensor. 
+
+**Examples of workgroup distribution with extended sg_map**
+
+The workgroup creates a tensor descriptor [128, 128] and distributes to 4 subgroups with `sg_layout` [2,2], and each subgroup gets `sg_data` [32,128]. The first dimension is split and distributed to subgroups in two rounds, and the second dimension is assigned as whole to multiple subgroups.
+
+```mlir
+   #sg_map_a = #xegpu.sg_map<sg_layout = [2, 2], sg_data = [32, 128], wi_layout=[1,16], wi_data = [1, 1], order = [1, 0]>
```

Comment:
By exposing sg_data as another degree of freedom, we allow nested wrap-around of lanes:  inside a sub-group tile, and over the entire tensor. 
Consider:
`#sg_map_a = #xegpu.sg_map<sg_layout = [2], sg_data = [32], wi_layout=[16], wi_data = [1], order = [1, 0]>
tensor_desc<128xf16, #sg_map_a>`

sg0:Lane0 will own elements 0, 16, 64, 80

Is there a motivation for expressing such complex walks, or can we just omit sg_data and have it inferred from wi_layout * wi_data ?

Reply:
Yes.  we use sg_data to support broadcast at wg to sg level, but "evenly distribute" at sg to wi level.  
Say, Matrix A [64, 64] at wg level, we have 4 sg in 2x2 layout, each sg takes [32,64] (note the second dimension is broadcast to 2 layout in a row). 
The map needs to be - 
#sg_map_a = #xegpu.sg_map<sg_layout = [2, 2], sg_data = [32, 64], wi_layout=[1, 16], wi_data = [1, 1], order = [1, 0]>

Reply:
> but "evenly distribute" at sg to wi level
This might be confusing. Might as well just allow same broadcast/wrap-around semantics within a sub-group.